File: 백준/Gold/1806. 부분합/부분합.py
import sys
input = sys.stdin.readline

n, s = map(int, input().split())
nums = list(map(int, input().split()))

left = 0
right = 0
sum_nums = nums[0]
answer = sys.maxsize
while left <= right:
    if sum_nums < s:
        right += 1
        if right == n:
            break
        sum_nums += nums[right]
    else:
        answer = min(answer, right - left + 1)
        sum_nums -= nums[left]
        left += 1

if answer == sys.maxsize:
    print(0)
else:
    print(answer)

File: 백준/Gold/1806. 부분합/부분합.js
const input = require('fs').readFileSync(process.platform === 'linux' ? '/dev/stdin' : __dirname + '/input.txt').toString().split('\n')

const [N, S] = input[0].split(' ').map(Number)
const arr = input[1].split(' ').map(Number)

let answer = N + 1
let left = 0
let right = 0
let sum = arr[left]
while(left <= right && right < N) {
  if (sum < S) {
    right++
    sum += arr[right]
  } else {
    sum -= arr[left]
    answer = Math.min(answer, right - left + 1)
    left++
  }
}

console.log(answer === N + 1? 0 : answer)